frontend/src/metabase/visualizations/components/ChartSettings/types.ts

```diff
@@ -1,15 +1,42 @@
+import type { ComponentType } from "react";
+
+import type {
+  VisualizationSettingDefinition,
+  WidgetComponentDef,
+} from "metabase/visualizations/types";
 import type Question from "metabase-lib/v1/Question";
 import type { Series, VisualizationSettings } from "metabase-types/api";
 
-// this type is not full, we need to extend it later
-export type Widget = {
-  id: string;
+export type Widget<
```

Comment:
this type is modeled after what is found in `getSettingWidget` in `frontend/src/metabase/visualizations/lib/settings.js`